components/raftstore/src/store/worker/split_controller.rs

```diff
@@ -639,6 +858,49 @@ impl AutoSplitController {
             top_qps.push(qps);
         }
 
+        // Check if the top CPU usage region could be split.
+        // TODO: avoid unnecessary split by introducing the feedback mechanism from PD.
+        if !top_cpu_usage.is_empty() && !is_grpc_poll_busy {
+            // Calculate by using the latest CPU usage.
+            top_cpu_usage.sort_by(|a, b| {
+                let cpu_usage_a = self.recorders.get(a).unwrap().cpu_usage;
+                let cpu_usage_b = self.recorders.get(b).unwrap().cpu_usage;
+                cpu_usage_b.partial_cmp(&cpu_usage_a).unwrap()
+            });
+            let region_id = top_cpu_usage[0];
+            let recorder = self.recorders.get_mut(&region_id).unwrap();
+            if recorder.hottest_key_range.is_some() {
+                split_infos.push(SplitInfo::with_start_end_key(
+                    region_id,
+                    recorder.peer.clone(),
+                    recorder
+                        .hottest_key_range
+                        .as_ref()
+                        .unwrap()
+                        .start_key
+                        .clone(),
+                    recorder.hottest_key_range.as_ref().unwrap().end_key.clone(),
+                ));
+                LOAD_BASE_SPLIT_EVENT
+                    .with_label_values(&[READY_TO_SPLIT_CPU_TOP])
+                    .inc();
+                info!("load base split region";
+                    "region_id" => region_id,
+                    "start_key" => log_wrappers::Value::key(&recorder.hottest_key_range.as_ref().unwrap().start_key),
+                    "end_key" => log_wrappers::Value::key(&recorder.hottest_key_range.as_ref().unwrap().end_key),
+                    "cpu_usage" => recorder.cpu_usage,
+                );
+            } else {
+                LOAD_BASE_SPLIT_EVENT
+                    .with_label_values(&[UNABLE_TO_SPLIT_CPU_TOP])
+                    .inc();
+            }
+        }
+        // Clean up the recorders.
+        for region_id in top_cpu_usage {
```

Comment:
clear?

Reply:
There are other regions in recorders that are not yet ready, so you cannot use clear directly.

Reply:
Oh, right.